src/pygeoprocessing/routing/routing.pyx

```diff
@@ -923,7 +955,7 @@ def fill_pits(
                             break
 
                         if pit_mask_managed_raster.get(
-                                xi_n, yi_n) == feature_id:
+                                xi_n, yi_n) != mask_nodata:

```

Comment:
`mask_nodata` is better here because it might be possible `feature_id` for `xi_n`, `yi_n` is not `feature_id` but has still been visited and has a different `feature_id`?

Reply:
Correct. That used to not be the case when you couldn't stop on the iteration of a fill region part way through.